synapse/api/auth.py

```diff
@@ -662,7 +664,9 @@ def check_in_room_or_world_readable(self, room_id, user_id):
             ):
                 return Membership.JOIN, None
             raise AuthError(
-                403, "Guest access not allowed", errcode=Codes.GUEST_ACCESS_FORBIDDEN
```

Comment:
this was never anything to do with guest access, so the error message was wrong and confusing; I've fixed it up while I'm here.